### DISCUSSION TITLE
How can I specify a branch when using "vercel deploy --prebuilt"?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [vercel/community#3660] [Help]
### Summary

I'm building a custom CLI, that makes it possible to deploy to Vercel. The project is also connected to Github.

My question is, from my local machine, using the Vercel CLI, how do I deploy `--prebuilt` to the stage preview branch?

Thanks 🙏

### Steps to Reproduce

If I just run `vercel deploy --prebuilt` I get a "new" preview environment, disconnected from "stage".

**Comment by TooTallNate** [comment]
Are you wanting to have the prebuilt deployment have a URL that matches the preview URL that would be assigned when you pushed to that same branch? If yes, then CLI deployments do not currently have that behavior automatically. You would have to run `vc alias <prebuilt deployment URL> <preview branch URL>` manually after the prebuilt deployment completes.

**Reply by alexbjorlig** [reply]
Thanks for the quick feedback. Ok so it would be something like `vercel alias set <prebuilt deployment URL> <stage branch URL>`

And juts to understand this, what is the "preview branch URL"? Would that be my "DNS" configured domain (stage.<domain>) or something else?

**Reply by TooTallNate** [reply]
The preview branch URL can be whatever you desire, but if you wish to emulate the behavior of the Git integration's preview URL, you would have to do some calculations. The Vercel Azure DevOps integration does something similar, so maybe [the code here](https://github.com/vercel/vercel-azure-devops-extension/blob/378e119ddd41e16cad818d8e67ff297f70d98b2d/vercel-deployment-task-source/src/index.ts#L220-L234) would be interesting for you.

**Reply by alexbjorlig** [reply]
Ok this seems a bit advanced 😅 Is it something Vercel is planning to add support for in the near future?

**Reply by rvanlaarhoven** [reply]
@TooTallNate , thanks for that code example! I was struggling to get the alias to work, since I kept running into `You don't have access to the domain` errors while providing the alias in this `[project-name]-[branch-name]-[org-name].vercel.app` format. The `--scope` flag did the trick.
Not sure if this is actually a bug though? The `pull` & `deploy` commands work fine without that flag when the `VERCEL_ORG_ID` env var is set, so I'd expect the `alias` command to work the same way.
